src/attributes.jl

```diff
@@ -86,16 +88,32 @@ function create_attribute(parent::Union{File,Object}, name::AbstractString, data
     end
     return obj, dtype
 end
-function create_attribute(parent::Union{File,Object}, name::AbstractString, dtype::Datatype, dspace::Dataspace)
-    attrid = API.h5a_create(checkvalid(parent), name, dtype, dspace, _attr_properties(name), API.H5P_DEFAULT)
+function create_attribute(
+    parent::Union{File,Object},
+    name::AbstractString,
+    dtype::Datatype,
+    dspace::Dataspace,
+)
+    attrid = API.h5a_create(
+        checkvalid(parent),
+        name,
+        dtype,
+        dspace,
+        _attr_properties(name),
+        API.H5P_DEFAULT,
+    )
     return Attribute(attrid, file(parent))
 end
 
 # generic method
 write_attribute(attr::Attribute, memtype::Datatype, x) = API.h5a_write(attr, memtype, x)
 # specific methods
 function write_attribute(attr::Attribute, memtype::Datatype, x::AbstractArray)
-    length(x) == length(attr) || throw(ArgumentError("Invalid length: $(length(x)) != $(length(attr)), for attribute \"$(name(attr))\""))
+    length(x) == length(attr) || throw(
```

Comment:
This  shouldn't be a  one liner

Reply:
You mean we should manually convert to plain `if ... end`, right ?

Reply:
yep

Reply:
Let's do this after this pull request